### ISSUE TITLE
Feature request: Nested `@timeit` blocks

### ISSUE THREAD

**Issue by fredrikekre** [KristofferC/TimerOutputs.jl#3]
Nested `@timeit` blocks similar to nested `@testset`'s. Currently:

```julia
julia> @timeit "outer" begin
           sleep(0.1)
           for i in 1:5
               @timeit "inner" sleep(0.1)
           end
       end

julia> print_timer()
+--------------------------------+-----------+--------+---------+
| Wall time elapsed since start  |    16.9 s |        |         |
|                                |           |        |         |
| Section              | n calls | wall time | % tot  | % timed |
+----------------------+---------+-----------+--------+---------+
| outer                |       1 |   0.607 s |   3.6% |   54.5% |
| inner                |       5 |   0.506 s |     3% |   45.5% |
+----------------------+---------+-----------+--------+---------+
```

Perhaps it could be printed with indentation like this:

```julia
julia> print_timer()
+--------------------------------+-----------+--------+---------+
| Wall time elapsed since start  |    16.9 s |        |         |
|                                |           |        |         |
| Section              | n calls | wall time | % tot  | % timed |
+----------------------+---------+-----------+--------+---------+
| outer                |       1 |   0.607 s |   3.6% |   54.5% |
|   inner              |       5 |   0.506 s |     3% |   45.5% |
+----------------------+---------+-----------+--------+---------+
```

and perhaps with a keyword, like

```julia
julia> print_timer(verbose=false)
+--------------------------------+-----------+--------+---------+
| Wall time elapsed since start  |    16.9 s |        |         |
|                                |           |        |         |
| Section              | n calls | wall time | % tot  | % timed |
+----------------------+---------+-----------+--------+---------+
| outer                |       1 |   0.607 s |   3.6% |   54.5% |
+----------------------+---------+-----------+--------+---------+

julia> print_timer(verbose=true) # default to true
+--------------------------------+-----------+--------+---------+
| Wall time elapsed since start  |    16.9 s |        |         |
|                                |           |        |         |
| Section              | n calls | wall time | % tot  | % timed |
+----------------------+---------+-----------+--------+---------+
| outer                |       1 |   0.607 s |   3.6% |   54.5% |
|   inner              |       5 |   0.506 s |     3% |   45.5% |
+----------------------+---------+-----------+--------+---------+
```

**Comment by KristofferC** [comment]
```
julia> @timeit to "foo" begin 
         sleep(0.5)
         @timeit to "bar" begin
             sleep(0.5)
             @timeit to "bur" begin
                 sleep(0.25)
            end
       end
       @timeit to "kek" sleep(0.25)
       end
       @timeit to "laa" sleep(0.1)

julia> to
 ══════════════════════════════════════════════════════════════════════════════
                                         Time                 Allocations      
                               ──────────────────────   ───────────────────────
  Total/percentage measured:        1.61 s / 0.02%          6.03KiB / 0.01%    
                                                                
  Section              ncalls    time   %tot  %timed      alloc   %tot  %alloc
 ──────────────────────────────────────────────────────────────────────────────
  laa                       1    101ms  6.30%  6.30%      496  B  8.03%  8.03%
  foo                       1   1.51 s  93.7%  93.7%     5.55KiB  92.0%  92.0%
    bar                     1    753ms  46.9%  46.9%     2.16KiB  35.8%  35.8%
      bur                   1    251ms  15.6%  15.6%      496  B  8.03%  8.03%
    kek                     1    251ms  15.6%  15.6%      496  B  8.03%  8.03%
 ══════════════════════════════════════════════════════════════════════════════
```

**Comment by KristofferC** [comment]
Perhaps guiding lines for the indents would be good..?

**Comment by KristofferC** [comment]
Or different colors

**Comment by fredrikekre** [comment]
Use them Crayons

**Comment by KristofferC** [comment]
This is pretty cool:

```
julia> to["foo"]
 ══════════════════════════════════════════════════════════════════════════════
                                         Time                 Allocations      
                               ──────────────────────   ───────────────────────
  Total/percentage measured:        1.02 s / 0.01%           144KiB / 0.16%    
                                                                
  Section              ncalls    time   %tot  %timed      alloc   %tot  %alloc
 ──────────────────────────────────────────────────────────────────────────────
  bar                       1    766ms  75.3%  75.3%      144KiB  99.7%  99.7%
    bur                     1    251ms  24.7%  24.7%      496  B  0.34%  0.34%
  kek                       1    251ms  24.7%  24.7%      496  B  0.34%  0.34%
 ══════════════════════════════════════════════════════════════════════════════
```

**Comment by fredrikekre** [comment]
Make a WIP PR?

**Comment by fredrikekre** [comment]
Fixed by #7
